core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1107,7 +1107,7 @@ class KafkaApis(val requestChannel: RequestChannel,
             .map(_.leader)
 
           coordinatorEndpoint match {
-            case Some(endpoint) if !endpoint.isEmpty =>
+            case Some(endpoint) if endpoint != null && !endpoint.isEmpty =>
```

Comment:
It's a little annoying that we have to handle both `null` and `None`. Any reason not to use `flatMap` as suggested offline?

```scala
.flatMap(p => Option(p.leader))
```
Or something like that.